sdcm/nemesis.py

```diff
@@ -5151,6 +5163,95 @@ def disrupt_disable_binary_gossip_execute_major_compaction(self):
+        from target node passed to cluster
+        """
+        if not self.target_node.raft.is_consistent_topology_changes_enabled:
+            raise UnsupportedNemesis("Raft feature: consistent-topology-changes is not enabled")
+        if self._is_it_on_kubernetes():
+            raise UnsupportedNemesis("Skip test for K8S because no supported yet")
+        keyspace_name = "banned_keyspace"
+        table_name = "table1"
+
+        def prepare_test_keyspace(node):
+            """ Create new table for verification missing data
+
+            Create new keyspace and table which allow to
+            verify that no operations come from banned node
+            """
+
+            with self.cluster.cql_connection_exclusive(node=node) as session:
+                tablets_config = TabletsConfiguration(
+                    enabled=True) if is_tablets_feature_enabled(session=session) else None
+
+            self.tester.create_keyspace(keyspace_name, replication_factor=3, tablets_config=tablets_config)
+            self.tester.create_table(name=table_name, keyspace_name=keyspace_name, key_type="bigint",
+                                     columns={"name": "text"})
+
+        simulate_node_unavailability = node_operations.block_scylla_ports if use_iptables else node_operations.stop_scylla_with_sigstop
+        with self.run_nemesis(node_list=self.cluster.nodes,
+                              nemesis_label=f"Running {simulate_node_unavailability.__name__}") as working_node, ExitStack() as stack:
+            target_host_id = self.target_node.host_id
+
+            stack.callback(self._remove_node_add_node, verification_node=working_node, node_to_remove=self.target_node,
+                           remove_node_host_id=target_host_id)
+
+            prepare_test_keyspace(self.target_node)
+            with simulate_node_unavailability(self.target_node):
+                # target node stopped by Contextmanger. Wait while its status will be updated
+                wait_for(node_operations.is_node_seen_as_down, timeout=600, throw_exc=True,
+                         down_node=self.target_node, verification_node=working_node, text=f"Wait other nodes see {self.target_node.name} as DOWN...")
+                self.log.debug("Remove node %s : hostid: %s with blocked scylla from cluster",
+                               self.target_node.name, target_host_id)
+                working_node.run_nodetool(f"removenode {target_host_id}", retry=3)
+                self.log.debug("Wait while all node update topology status")
+                wait_for(node_operations.is_node_removed_from_cluster, timeout=300, throw_exc=True,
+                         removed_node=self.target_node, verification_node=working_node, text=f"Waite other nodes mark node {self.target_node.name} as REMOVED...")
+                # Run read barrier on working node to validate that it update raft topology state
+                working_node.raft.call_read_barrier()
```

Comment:
We don't have execute the read barrier here. You can remove it.

It is true that we must ensure all live nodes apply the Raft entry responsible for banning the removed node before we test that the removed node is actually banned. Read barrier is a mechanism that allows it.

However, in the current implementation of Scylla, we ban the node being removed at the beginning of the removal procedure, so that it doesn't disrupt us if it comes back to live. We commit the Raft entry responsible for banning the node early and wait until all nodes apply it by using a global read barrier.

We didn't implement the Raft-based removenode this way initially. In the old (and incorrect) implemantion, the removed node was banned at the end of the removal procedure and we didn't execute any read barrier after that. With such implementation, executing read barriers on all nodes in this test would be necessary.

Reply:
```python
wait_for(node_operations.is_node_removed_from_cluster, timeout=300, throw_exc=True,
         removed_node=self.target_node, verification_node=working_node, text=f"Waite other nodes mark node {self.target_node.name} as REMOVED...")
```
is also unnecessary. The test should always pass without it. When `nodetool removenode` finishes, all nodes should have already banned the removed node.